recursion/gnark/sp1/utils.go

```diff
@@ -124,3 +130,134 @@ func WriteToFile(filename string, writerTo io.WriterTo) error {
 
 	return nil
 }
+
+// Helper function to check if a file exists.
+func fileExists(filePath string) bool {
+	_, err := os.Stat(filePath)
+	return !os.IsNotExist(err)
+}
+
+// LoadCircuit checks if the necessary circuit files are in the specified data directory,
+// downloads them if not, and loads them into memory.
+func LoadCircuit(dataDir, circuitType string) (constraint.ConstraintSystem, groth16.ProvingKey, groth16.VerifyingKey, error) {
```

Comment:
Note: This was moved from `recursion/gnark/server/load.go` as `sp1` should have the utils that are used across `sp1` and `server`, given that we'll deprecate `server` soon.